architectures/BDR-Always-ON/commands/upgrade.yml

```diff
@@ -33,6 +33,11 @@
   environment: "{{ target_environment }}"
   hosts: all
   tasks:
+  - include_role: name=postgres/facts
+    vars:
+      pgdata_initialised: true
+    when: >
+      'postgres' in role and cluster_facts is not defined
```

Comment:
The earlier play includes init.yml, which does cluster_discovery already. Under what circumstances is this additional  invocation required?

Reply:
I agree but I found that cluster_facts isn't being picked up without this task.

Reply:
I can retest and verify if I still observe this behavior.